### PR TITLE
Sum up all the votes and voters for the entire election for PDF

### LINKED ISSUE #224: Combine input for Typst PDF generation in backend
Add up results of all polling stations to the total counts of the municipality.

- Assert that there are no errors in the data entry (run validation, check for no errors).
- Use `voters_recounts` instead of `voters_counts` when recounted.

### PR DESCRIPTION
**Pull request by rnijveld**
Fixes #224 

The tests right now are relatively barebones, but at least we test some successful additions. I would suggest to at add much more significant testing as an issue for a later time (soonish)

### CODE REVIEW

File: backend/src/pdf_gen/models/model_na_31_2.rs [hunk 1]
```diff
@@ -41,6 +45,66 @@ impl ModelNa31_2Summary {
...
+                        })
+                        .collect(),
+                });
+            }
+        }
+
+        // loop over results and add them to the running total
+        for (polling_station, result) in results {
+            // add voters and votes to the total
+            totals.voters_counts += &result.voters_counts;
```
Comment (praseodym) [inline backend/src/pdf_gen/models/model_na_31_2.rs:77]: If recounted, we need to use the recounted voters counts instead.

**Review by praseodym — CHANGES_REQUESTED** [review]

File: backend/src/pdf_gen/models/model_na_31_2.rs (hunk 1 above)
Comment (jorisleker) [reply backend/src/pdf_gen/models/model_na_31_2.rs:77]: Adding up results for the entire election is the core functionality of our software. 

Feels a bit odd to do the addition inside `/pfd_gen` and inside a specific model. Shouldn't this be in a dedicated module (that is also reused for generating eml's?

Comment (praseodym) [reply backend/src/pdf_gen/models/model_na_31_2.rs:77]: Agreed! I don't think its necessary to move the code at this point already, but we should keep it in mind.

Comment (jorisleker) [reply backend/src/pdf_gen/models/model_na_31_2.rs:77]: Maybe put in into a ticket, so we don't forget?

Comment (jschuurk-kr) [reply backend/src/pdf_gen/models/model_na_31_2.rs:77]: I'd hope/expect that as soon as we do a calculation that's not for an Na31-2, moving that code is the obvious choice, without needing a ticket to remember.

**Comment by jschuurk-kr** [comment]
When I run it, pdf generation prints some debugging info: the data, some durations, and the number of warnings. Do we want to keep that for now?

(note: the relevant code is not in scope of this PR, it was added in [PR #307](https://github.com/kiesraad/abacus/pull/307))

File: backend/src/polling_station/structs.rs [hunk 2]
```diff
@@ -389,6 +397,15 @@ pub struct VotersCounts {
...
 
+impl AddAssign<&VotersCounts> for VotersCounts {
+    fn add_assign(&mut self, other: &Self) {
+        self.poll_card_count += other.poll_card_count;
+        self.proxy_certificate_count += other.proxy_certificate_count;
+        self.voter_card_count += other.voter_card_count;
+        self.total_admitted_voters_count += other.total_admitted_voters_count;
+    }
+}
+
```
Comment (Lionqueen94) [inline backend/src/polling_station/structs.rs:408]: There needs to be an implementation for adding votersRecounts to VotersCounts as well

File: backend/src/pdf_gen/models/model_na_31_2.rs (hunk 1 above)
Comment (jorisleker) [reply backend/src/pdf_gen/models/model_na_31_2.rs:77]: 🤞

Comment (rnijveld) [reply backend/src/pdf_gen/models/model_na_31_2.rs:77]: Moving it to a more generic place comes naturally as more functionality is added I would say. As a part of KISS (and for the little time we currently have) I would try and prevent doing more than is necessary for the functionality we're currently adding. I'd imagine that the EML output and adding other models will have other consequences that we currently can't foresee anyway.

File: backend/src/pdf_gen/models/model_na_31_2.rs [hunk 3]
```diff
@@ -41,6 +45,66 @@ impl ModelNa31_2Summary {
...
+            totals
+                .differences_counts
+                .add_polling_station_results(polling_station, &result.differences_counts);
+
+            // if this polling station was recounted, add it to the list
+            if result.recounted {
+                totals
+                    .recounted_polling_stations
+                    .push(polling_station.number);
+            }
```
[the remark this answers is not in the document]
Comment (praseodym) [reply backend/src/pdf_gen/models/model_na_31_2.rs:90]: It is used in the report:
Screenshot 2024-09-12 at 11 40 14

**Review by jorisleker — APPROVED** [review]
Tested, download now works on macos in chrome, safari and firefox

File: backend/src/pdf_gen/models/model_na_31_2.rs (hunk 1 above)
Comment (Lionqueen94) [reply backend/src/pdf_gen/models/model_na_31_2.rs:77]: I see what you mean @rnijveld , but the section for recounts and differences is unfortunately the same, see:
Screenshot from 2024-09-12 12-27-09
Screenshot from 2024-09-12 12-27-26

We could add some more information in the comments though, what do you think @praseodym @jschuurk-kr ?

Comment (praseodym) [reply backend/src/pdf_gen/models/model_na_31_2.rs]: Improved in 3ddc7e0de5c21401cb8c36991b427a1609c37e93, let me know what you think!

File: backend/src/pdf_gen/models/model_na_31_2.rs [hunk 4]
```diff
@@ -81,4 +170,280 @@ impl SumCount {
...
+
+    pub fn add(&mut self, polling_station: &PollingStation, count: Count) {
+        if count > 0 {
+            self.count += count;
+            self.polling_stations.push(polling_station.number);
+        }
+    }
+}
+
+#[cfg(test)]
```
[the remark this answers is not in the document]
Comment (praseodym) [reply backend/src/pdf_gen/models/model_na_31_2.rs:203]: Added in 3ddc7e0de5c21401cb8c36991b427a1609c37e93.

**Review by Lionqueen94 — APPROVED** [review]

File: backend/src/pdf_gen/models/model_na_31_2.rs (hunk 1 above)
Comment (rnijveld) [reply backend/src/pdf_gen/models/model_na_31_2.rs]: Looks good to me! <sub>I can't give approval on the PR though because I'm the author of it</sub>

**Review by Lionqueen94 — APPROVED** [review]

**Review by praseodym — APPROVED** [review]

**Pull request merged**